### DISCUSSION TITLE
Add NODE_OPTIONS in launch.json, bug get empty string

### DISCUSSION THREAD

**Discussion by qiufengiii** [microsoft/vscode-discussions#1347] [Extension Development QnA] [answered]
I'm trying to add NODE_OPTIONS to extension node env, this is my launch.json
<img width="538" alt="image" src="https://github.com/user-attachments/assets/ce65d3c1-f3f1-4437-bcec-9bc9dadeefa2">

but only get empty string. What should i do？
<img width="213" alt="image" src="https://github.com/user-attachments/assets/3fecb082-b24d-4405-877f-36fc56aaef20">

**Comment by isidorn** [comment]
Maybe @connor4312 might be able to help.
Maybe the Extension Host process does not respect all the passed flags.

**Comment by robconery** [comment]
Hi qiufengiii - what version of VS Code are you using, and which operating system? The `env` flag should work, and I verified your configuration using build 1.92.2. Perhaps check the build to be sure if it's the latest as well. The `NODE_OPTIONS` environment setting should have quite a few settings in it.

Finally - what type of application is this?

**Reply by olguzzar** [reply]
Hi @qiufengiii, could you please respond to the above questions so we can better assist you? If we don't receive more information in the next week, we will close this discussion due to inactivity (but can reopen upon request!)

**Comment by connor4312** [comment, marked as answer]
NODE_OPTIONS are not able to be set for the VS Code extension host.

**Comment by James4Ever0** [comment]
> NODE_OPTIONS are not able to be set for the VS Code extension host.

Will altering the code of electron make `NODE_OPTIONS` working for VSCode extrnsion host?

Do I need to both set `NODE_OPTIONS` and `VSCODE_NODE_OPTIONS` when running VSCode, or do I need to modify `extensionHostProcess`?

https://github.com/electron/electron/issues/38875

---

It seems a hack but works:

```bash
# under the VSCode installation folder, ("Microsoft VS Code" on Windows)

sed -i 's/process.env.ELECTRON_RUN_AS_NODE="1"/\0,process.env.NODE_OPTIONS="<your_options>"/g' resources/app/out/vs/workbench/api/node/extensionHostProcess.js
```
